### PR TITLE
WIP: Add warn about old driver version (hyperkit, kvm)

### PR DESCRIPTION
**Pull request by RA489**
Add warn about old driver version (hyperkit, kvm)
fixes https://github.com/kubernetes/minikube/issues/4658

### CODE REVIEW

File: pkg/minikube/problem/err_map.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ var vmProblems = map[string]match{
 		URL:    "https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver",
 		Issues: []int{1926, 4206},
 	},
+	"OLD_HYPERKIT_VERSION": {
+		Regexp: re(`Too old hyperkit version found`),
+		Advice: "too old hyperkit version. Jump forward to new valid hyperkit version as a workaround.",
+		Issues: []int{4658},
```
Comment (josedonizetti) [inline pkg/minikube/problem/err_map.go]: Add URL to documentation on how to upgrade.

File: pkg/minikube/problem/err_map.go [hunk 2]
```diff
@@ -119,6 +124,11 @@ var vmProblems = map[string]match{
 		URL:    "http://mikko.repolainen.fi/documents/virtualization-with-kvm",
 		Issues: []int{2991},
 	},
+	"OLD_KVM_VERSION": {
+		Regexp: re(`Too old KVM version found`),
+		Advice: "too old KVM version. Jump forward to new valid KVM version as a workaround.",
```
Comment (josedonizetti) [inline pkg/minikube/problem/err_map.go]: Add URL to documentation on how to upgrade.

**Comment by josedonizetti** [comment]
@RA489 I added two comments, but I don't understand how this fixes the issue. For me, the problem package would display a message based on an existing error that matches the regex. Although I never saw those drivers message before `Too old KVM/Hyperkit version found`, if they already exist, I can close the PR I opened to check it https://github.com/kubernetes/minikube/pull/4676, which will probably won't be necessary. Thanks :)

File: pkg/minikube/problem/err_map.go (hunk 1 above)
Comment (tstromberg) [reply pkg/minikube/problem/err_map.go]: Add this doc: https://github.com/kubernetes/minikube/blob/master/docs/drivers.md

Comment (RA489) [reply pkg/minikube/problem/err_map.go]: Done!

File: pkg/minikube/problem/err_map.go (hunk 2 above)
Comment (RA489) [reply pkg/minikube/problem/err_map.go]: @josedonizetti done!

File: pkg/minikube/problem/err_map.go (hunk 1 above)
Comment (RA489) [reply pkg/minikube/problem/err_map.go]: done

File: pkg/minikube/problem/err_map.go (hunk 2 above)
Comment (RA489) [reply pkg/minikube/problem/err_map.go]: done

File: pkg/minikube/problem/err_map.go [hunk 3]
```diff
@@ -36,6 +36,11 @@ var vmProblems = map[string]match{
 		URL:    "https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver",
 		Issues: []int{1926, 4206},
 	},
+	"OLD_HYPERKIT_VERSION": {
+		Regexp: re(`Too old hyperkit version found`),
```
[the remark this answers is not in the document]
Comment (RA489) [reply pkg/minikube/problem/err_map.go]: Done!

File: pkg/minikube/problem/err_map.go [hunk 4]
```diff
@@ -36,6 +36,12 @@ var vmProblems = map[string]match{
 		URL:    "https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver",
 		Issues: []int{1926, 4206},
 	},
+	"OLD_HYPERKIT_VERSION": {
+		Regexp: re(`Add warn about old driver version (hyperkit, kvm)`),
+		Advice: "Too old hyperkit version. Jump forward to new valid hyperkit version as a workaround.",
+		URL:	"https://github.com/kubernetes/minikube/blob/master/docs/drivers.md"
```
Comment (josedonizetti) [inline pkg/minikube/problem/err_map.go]:
```suggestion
		URL:	"https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-upgrade",
```

There's a specific link for hyperkit upgrade. Also missing a comma at the end.

File: pkg/minikube/problem/err_map.go [hunk 5]
```diff
@@ -119,6 +125,12 @@ var vmProblems = map[string]match{
 		URL:    "http://mikko.repolainen.fi/documents/virtualization-with-kvm",
 		Issues: []int{2991},
 	},
+	"OLD_KVM_VERSION": {
+		Regexp: re(`Too old KVM version found`),
+		Advice: "too old KVM version. Jump forward to new valid KVM version as a workaround.",
+		URL:	"https://github.com/kubernetes/minikube/blob/master/docs/drivers.md"
```
Comment (josedonizetti) [inline pkg/minikube/problem/err_map.go]:
```suggestion
		URL:	"https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#kvm2-upgrade",
```

There's a specific link for kvm upgrade. Also missing a comma at the end.

File: pkg/minikube/problem/err_map.go [hunk 6]
```diff
@@ -36,6 +36,12 @@ var vmProblems = map[string]match{
 		URL:    "https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver",
 		Issues: []int{1926, 4206},
 	},
+	"OLD_HYPERKIT_VERSION": {
+		Regexp: re(`Add warn about old driver version (hyperkit, kvm)`),
```
Comment (josedonizetti) [inline pkg/minikube/problem/err_map.go]: Which message is this regexp matching? Do you have an example you a show of where this message is displayed to the user?

File: pkg/minikube/problem/err_map.go [hunk 7]
```diff
@@ -119,6 +125,12 @@ var vmProblems = map[string]match{
 		URL:    "http://mikko.repolainen.fi/documents/virtualization-with-kvm",
 		Issues: []int{2991},
 	},
+	"OLD_KVM_VERSION": {
+		Regexp: re(`Too old KVM version found`),
```
Comment (josedonizetti) [inline pkg/minikube/problem/err_map.go]: Which message is this regexp matching? Do you have an example you a show of where this message is displayed to the user?

**Comment by tstromberg** [comment]
Closing in preference to #4691

**Pull request closed without merging**